File: statsig-php/src/EvaluationTypes/ExperimentGroup.php
<?php

namespace Statsig\EvaluationTypes;

class ExperimentGroup
{
    public string $groupName;
    public string $ruleId;
    public string $idType;
    public array $returnValue;

    public function __construct(array $group)
    {
        $this->groupName = (string)($group['group_name'] ?? '');
        $this->ruleId = (string)($group['rule_id'] ?? '');
        $this->idType = (string)($group['id_type'] ?? '');
        $this->returnValue = $group['return_value'] ?? [];
    }
}

File: statsig-elixir/lib/statsig/experiment_group.ex
defmodule Statsig.ExperimentGroup do
  @moduledoc """
  A group within an experiment, exposing its name, rule id, id type, and
  return value without requiring a user evaluation.
  """

  defstruct [
    :group_name,
    :rule_id,
    :id_type,
    :return_value
  ]

  @type t :: %__MODULE__{
          group_name: String.t(),
          rule_id: String.t(),
          id_type: String.t(),
          return_value: %{optional(String.t()) => any()}
        }
end
